cmd/acr/purge.go

```diff
@@ -614,8 +617,65 @@ func dryRunPurge(ctx context.Context, acrClient api.AcrCLIClientInterface, login
 	return deletedTagsCount, deletedManifestsCount, nil
 }
 
+// getDoNotDeleteListFromMultiArchManifestWithTag checks if a manifest has tags and its media type support multiarch manifests.
+// If so, the dependent manifests are marked not to be deleted
+func getDoNotDeleteListFromMultiArchManifestWithTag(ctx context.Context, manifest acr.ManifestAttributesBase, doNotDelete set.Set[string], acrClient api.AcrCLIClientInterface, repoName string) error {
```

Comment:
WOW such a long name!
It looks like this function does not check if a manifest has a tag or not?
How about calling it something like `doNotDeleteDependantManifests`?